apimodels/task_log.go

```diff
@@ -54,149 +45,121 @@ type TaskLog struct {
+	case s < level.Info:
 		return LogDebugPrefix
 	default:
 		return LogInfoPrefix
 	}
 }
 
-// GetBuildloggerLogsOptions represents the arguments passed into
-// GetBuildloggerLogs function.
-type GetBuildloggerLogsOptions struct {
-	BaseURL       string `json:"_"`
-	TaskID        string `json:"-"`
-	TestName      string `json:"-"`
-	GroupID       string `json:"-"`
-	Execution     *int   `json:"-"`
-	PrintPriority bool   `json:"-"`
-	Tail          int    `json:"-"`
-	LogType       string `json:"-"`
+func GetPriority(prefix string) level.Priority {
+	switch prefix {
+	case LogErrorPrefix:
+		return level.Error
+	case LogWarnPrefix:
+		return level.Warning
+	case LogDebugPrefix:
+		return level.Debug
+	default:
+		return level.Info
+	}
 }
 
-// GetBuildloggerLogs makes request to Cedar for a specifc log and returns an
-// io.ReadCloser.
-func GetBuildloggerLogs(ctx context.Context, opts GetBuildloggerLogsOptions) (io.ReadCloser, error) {
-	usr := gimlet.GetUser(ctx)
-	if usr == nil {
-		return nil, errors.New("error getting user from context")
+// ReadLogToSlice returns a slice of log message pointers from a log iterator.
+func ReadLogToSlice(it log.LogIterator) ([]*LogMessage, error) {
+	var lines []*LogMessage
+	for it.Next() {
+		item := it.Item()
+		lines = append(lines, &LogMessage{
+			Severity:  GetSeverityMapping(item.Priority),
+			Message:   item.Data,
+			Timestamp: time.Unix(0, item.Timestamp),
+		})
 	}
-	getOpts := buildlogger.GetOptions{
-		Cedar: timber.GetOptions{
-			BaseURL:  fmt.Sprintf("https://%s", opts.BaseURL),
-			UserKey:  usr.GetAPIKey(),
-			UserName: usr.Username(),
-		},
-		TaskID:        opts.TaskID,
-		TestName:      opts.TestName,
-		GroupID:       opts.GroupID,
-		Execution:     opts.Execution,
-		PrintTime:     true,
-		PrintPriority: opts.PrintPriority,
-		Tail:          opts.Tail,
+	if err := it.Err(); err != nil {
+		return nil, errors.Wrap(err, "iterating log lines")
 	}
 
-	switch opts.LogType {
-	case TaskLogPrefix:
-		getOpts.Tags = []string{evergreen.LogTypeTask}
-	case SystemLogPrefix:
-		getOpts.Tags = []string{evergreen.LogTypeSystem}
-	case AgentLogPrefix:
-		getOpts.Tags = []string{evergreen.LogTypeAgent}
-	case AllTaskLevelLogs:
-		getOpts.Tags = []string{
-			evergreen.LogTypeTask,
-			evergreen.LogTypeSystem,
-			evergreen.LogTypeAgent,
-		}
-	}
-	logReader, err := buildlogger.Get(ctx, getOpts)
-
-	return logReader, errors.Wrapf(err, "getting logs for '%s' from Buildlogger, using Evergreen logger", opts.TaskID)
+	return lines, errors.Wrap(it.Close(), "closing log iterator")
 }
 
-// ReadBuildloggerToChan parses Cedar buildlogger log lines by message and
-// severity and reads into a channel.
-func ReadBuildloggerToChan(ctx context.Context, taskID string, r io.ReadCloser, lines chan<- LogMessage) {
-	var (
-		line string
-		err  error
-	)
+// StreamFromLogIterator streams log lines from the given iterator to the
+// returned log message channel. It is the responsibility to close the log
```

Comment:
"It is the responsibility to close the log" -- can you fix this wording?

Reply:
Done.